### DISCUSSION TITLE
Restricting the time selection on the date picker to only allow 5 min intervals.

### DISCUSSION THREAD

**Discussion by CCHV-GitHub** [Budibase/budibase#11812] [Help] [answered]
Hi,

New to budibase, enjoying trying to develop a tool on it.

I am trying to figure out how to restrict the date picker to only validate when the minutes are a multiple of 5, so 05, 10, 15, 20 etc. 
Unless theres a way to actually restrict the input so it only uses multiples of 5, either would work fine for my purpose.

![image](https://github.com/Budibase/budibase/assets/120619360/7f64122e-afc6-4521-9fae-8de013261796)

Would appreciate the help.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @CCHV-GitHub,

Sadly I don't think there is a way to use the in-built Budibase validation to do what you are after. So i had to go down a more unorthodox route. If you feel like you should be able to do validation similar to what you have described can you please create a feature request.

I think I achieved what you wanted. Instead of trying to go down the validation route. I opted to used the `update field value` action on change and rounded it to whatever number closer to 5/10 it will always opt to go for the higher number e.g. if you typed 04 it would round it to 05, if you did 07 it would round to 10. I hope this makes sense. The JS Code I wrote to achieve this is below;

```js
function roundToNearest5Minutes(dateTimeString) {
  const date = new Date(dateTimeString);
  const minutes = date.getMinutes();
  const remainder = minutes % 5;

  if (remainder === 0) {
    return date.toISOString();
  }

  const adjustment = 5 - remainder;
  
  date.setMinutes(minutes + adjustment, 0, 0);

  return date.toISOString();
}

const dateTimeString = $("Field Value");
const result = roundToNearest5Minutes(dateTimeString);
return result;
```

Here are some screenshots of how the component is setup below;
<img width="1512" alt="Screenshot 2023-09-20 at 12 15 31" src="https://github.com/Budibase/budibase/assets/126772285/b45590d9-486c-4557-a9d4-ea33f79781f2">
<img width="755" alt="Screenshot 2023-09-20 at 12 17 43" src="https://github.com/Budibase/budibase/assets/126772285/1719394d-f906-449e-aee8-1d8c75efe96b">

**Gif of the output**
![Screen Recording 2023-09-20 at 12 49 09](https://github.com/Budibase/budibase/assets/126772285/b3b2ecbe-6e92-486e-8aa1-f3383a174040)

I hope this is what you were after!

**Reply by melohagan** [reply]
Great answer! You could also explore building [Custom components](https://docs.budibase.com/docs/custom-component) if you want that exact UI control.
